Add Pinterest ads manager CTA

ianlin opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

Closes #252.

This PR adds a Pinterest ads manager CTA under catalog overview as described in #252.

### Screenshots:

#### When there are warnings

<img width="1278" alt="Screenshot 2021-11-23 at 17 54 57" src="https://user-images.githubusercontent.com/914406/143003725-a3819b37-1896-48ab-af4b-7c3ff1e9f49e.png">

#### When there are no warnings

<img width="1280" alt="Screenshot 2021-11-23 at 17 48 49" src="https://user-images.githubusercontent.com/914406/143003763-ea111381-105a-4f55-b381-c7f602f2d41b.png">

### Detailed test instructions:

1. Install Pinterest for WooCommerce and go through the onboarding wizard.
2. Go to `Marketing` -> `Pinterest`, then Click `Catalog`.
3. At the bottom of the `Overview` table, there is a Pinterest ads manager CTA

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

> Add - Pinterest ads manager CTA

ianlin committed:
Ignore vim's swap files in .gitignore

ianlin committed:
Update @wordpress/icons's version to >=5.0.0 <6.0.0

- Why >=5.0.0 and <6.0.0?
  - Because a new icon "trendingUp" will be used in this PR, and it's
    only added since 4.0.0.
    - https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/CHANGELOG.md#400-2021-05-20
  - 6.0.0 removes the `minus` icon which is used in @wp-g2/components module version 0.0.140
    - So there are errors when using 6.0.0
  - 5.0.0 is the newest version that has no errors

ianlin committed:
Add Pinterest ads manager CTA under catalog overview

ianlin commented on package.json:
```diff
@@ -21,7 +21,7 @@
     "@wordpress/html-entities": "^2.7.0",
     "@wordpress/i18n": "^3.11.0",
-    "@wordpress/icons": "^2.10.3",
+    "@wordpress/icons": ">=5.0.0 <6.0.0",
```
The reason why using `>=5.0.0 and <6.0.0`:

- Because a new icon `trendingUp` will be used in this PR, and it's only added [since 4.0.0][1].
- 6.0.0 removes the `minus` icon which is used in `@wp-g2/components` module version 0.0.140, so there are errors when using 6.0.0
- Therefore, 5.0.0 is the newest version that has no errors.

[1]: https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/CHANGELOG.md#400-2021-05-20

puntope commented on .gitignore:
```diff
@@ -22,3 +22,6 @@ assets/source/_maps
 pinterest-for-woocommerce.zip
+
+# vim
+*.swp
```
💅   Guess we can use here also [Global git ignore]( https://gist.github.com/subfuzion/db7f57fff2fb6998a16c) for this since it will impact all the repos.

Given that, I would group all of these (dev) rules if they are really necessary for the  `# Ignore node_modules and other local dev files` section to facilitate the reading.

This goes for the `vim` rule  and the previous ` .DS_Store` rule at the top.

ianlin replied:
Thanks for the global git ignore tip! I will add it locally and remove the one in `.gitignore` of this repo.

ianlin replied:
Fixed in ee48d0c.

puntope commented on assets/source/setup-guide/app/style.scss:
```diff
@@ -590,6 +590,18 @@ $root: ".woocommerce-setup-guide";
+		.components-card__footer {
+			flex-direction: row;
+
+			.link {
+				text-decoration: none;
+
+				&:hover {
+					text-decoration: underline;
+				}
+			}
```
💅 

I see that the rest of the links are actually underlined. Why we are removing the underline on this one is it a requirement?

Given that, In case it is a requirement:  

Why we are using this custom  `.link` selector instead of `.components-external-link` component's default?

ianlin replied:
Ha, I wasn't aware there is a default class used by `<ExternalLink>` component. 😅 TIL, thanks @puntope!

I will fix it soon and request your review again. 😄 

A side question: any tips to find out if there's a default class for a component? I've found two ways but would love to learn more.

- Open chrome devtool and inspect the target element, there's a class in the `<a>` tag, e.g.
```html
<a class="components-external-link link" href="https://ads.pinterest.com/" target="_blank" rel="external noreferrer noopener">
```
- In https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/external-link/index.js, it clearly uses `components-external-link` classname there.

ianlin replied:
Forgot to answered this:
> I see that the rest of the links are actually underlined. Why we are removing the underline on this one is it a requirement?

It's based on the design screenshot in #252, from the screenshot the link is not underlined so I assume it's a requirement.

puntope replied:
> A side question: any tips to find out if there's a default class for a component? I've found two ways but would love to learn more.

I think those options are good. Instead of going to Github. What I normally do is click on the import element in the IDE and then it shows me the src code of that import. 

<img width="1037" alt="Screenshot 2021-11-24 at 10 51 32" src="https://user-images.githubusercontent.com/5908855/143215285-b01225b2-caaa-4977-9b1a-3aace613a3a4.png">

ianlin replied:
Fixed in 8c68886.

puntope commented on assets/source/setup-guide/app/style.scss:
```diff
@@ -590,6 +590,18 @@ $root: ".woocommerce-setup-guide";
+		.components-card__footer {
+			flex-direction: row;
+
+			.link {
```
```suggestion
			.components-external-link {
```

puntope replied:
In case we don't need `.link`selector for some reason. I would go for `.components-external-link` default selector

ianlin replied:
Same as https://github.com/woocommerce/pinterest-for-woocommerce/pull/273#discussion_r755826043.

puntope commented on assets/source/catalog-sync/sections/SyncState.js:
```diff
@@ -30,6 +34,28 @@ const SyncState = () => {
+						{
+							adsManagerLink: (
+								<ExternalLink
+									className="link"
```
```suggestion
```

In case there is no specific reason to use a custom selector. I would go for the default.

ianlin replied:
Same as https://github.com/woocommerce/pinterest-for-woocommerce/pull/273#discussion_r755826043.

puntope reviewed (changes requested):
LGFM! ✅ 

🧪  Testing Process 

* Visual verification in my local machine. 
* No tests :( 

💅 

* Please check my comments regarding .gitignore and the CSS `link` selector. 
* I didn't approve because especially the one regarding `link` could be a requested change if there is no reason for using it. 
* Tests??? (I see react-testing-library + test runner are missing, I can help with that if needed)

ianlin commented:
@puntope 
> Tests??? (I see react-testing-library + test runner are missing, I can help with that if needed)

Totally agree that we should add tests and I'm ashamed of my laziness 🤣 Just seen you've created #275 regarding setting up testing library, thanks!
I would like to try that out, unless you've already started doing it.

ianlin committed:
Remove vim swap files in repo-wide .gitignore

Suggested by @puntope that one can use global .gitignore rather than
repo-wide .gitignore.
Ref: https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

puntope commented:
> I would like to try that out unless you've already started doing it.

I would also love to implement it. (Yes, I used it a lot but never did the implementation) 

What we can do is that I could implement the testing lib today and you can review it + implement the test on this one (which is a great way to have an intro to the library).

Besides that, if I see another project that needs this implementation I can ping you in order to try it out by yourself.

How does it sound?

ianlin commented:
> What we can do is that I could implement the testing lib today and you can review it + implement the test on this one (which is a great way to have an intro to the library).
Besides that, if I see another project that needs this implementation I can ping you in order to try it out by yourself.
How does it sound?

@puntope Perfect plan, thanks!

ianlin committed:
Use component's default classname for <ExternalLink>

findingsimple commented:
> ashamed of my laziness

No need, it wasn't set up before so it's understandable that you followed suit.

Thanks for starting to be more forward thinking with this project guys 🙇

puntope reviewed (approved):
✅   Perfect!
